frontend/store/collaterals.ts

```diff
@@ -69,6 +80,62 @@ export const actions = {
             commit('updateCollateral', updated);
         }
     },
+    async fetchCollateralStatuses({ commit, dispatch, rootGetters }: ActionContext<State, State>) {
+        const network = rootGetters['network/getMakerNetwork'];
+        if (!network) {
+            return;
+        }
+        for (const collateral of Object.values(COLLATERALS)) {
+            const tokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateral.symbol).catch(() => {
+                return undefined;
+            });
+            if (!tokenAddress) {
+                continue;
+            }
+            await dispatch('wallet/fetchCollateralVatBalance', collateral.ilk, { root: true });
+            await dispatch('authorizations/fetchCollateralAuthorizationStatus', collateral.ilk, { root: true });
+            const isAuthorized = rootGetters['authorizations/collateralAuthorizations'].includes(collateral.ilk);
+            const balance = rootGetters['wallet/collateralVatBalanceStore'][collateral.ilk];
+            commit('setCollateralStatus', {
+                type: collateral.ilk,
+                symbol: collateral.symbol,
+                address: tokenAddress,
+                isAuthorized,
+                balance,
+            });
+        }
+    },
+    async fetchCollateralStatus(
+        { commit, dispatch, rootGetters }: ActionContext<State, State>,
+        collateralType: string
+    ) {
+        const network = rootGetters['network/getMakerNetwork'];
+        if (!network) {
+            return;
+        }
+        const collateral = Object.values(COLLATERALS).find(collateral => collateral.ilk === collateralType);
+        if (!collateral) {
+            console.error('Unknown collateral:', collateralType);
+            return;
+        }
+        const tokenAddress = await getTokenAddressByNetworkAndSymbol(network, collateral.symbol).catch(() => {
+            return undefined;
+        });
+        if (!tokenAddress) {
+            return;
+        }
+        await dispatch('wallet/fetchCollateralVatBalance', collateralType, { root: true });
+        await dispatch('authorizations/fetchCollateralAuthorizationStatus', collateralType, { root: true });
+        const isAuthorized = rootGetters['authorizations/collateralAuthorizations'].includes(collateralType);
+        const balance = rootGetters['wallet/collateralVatBalanceStore'][collateralType];
```

Comment:
You don't need to take it from one store here and put it in another store. You can do it at the time you return that data (inside the getter you can join data from different places). See next comment